README.md

```diff
@@ -1,2 +1,418 @@
+
+**`<SearchForm.View model={model.form} />`** shows View of `SearchForm` for a model `form`.
+
+As you can see, to nest one block to another you have to nest all three parts of a block separately: model, logic, view. It is a bit overhead, but it makes nesting very flexible. For example, you can use a Logic of one block and compelely replace a View of this block to something more suitable in some concrete situation. Or you can create a model for a child block in some specific way.
+
+
+### Multiple counter problem
+What if each item should have some specific logic, like a counter? In MJS that is not a problem:
+```js
+// ResultsItem.js
+import { Cmd } from 'mangojuice-core';
+
+export const createModel = (text) => ({
+  counter: 0,
+  text
+});
+export const Logic = {
+  name: 'ResultsItem',
+
+  @Cmd.update
+  Increment({ model }, amount) {
+    return { counter: model.counter + amount };
+  }
+};
+export const View = ({ model }) => (
+  <div>
+    {model.text}<br />
+    <button onClick={Logic.Increment(1)}>+</button>
+    {model.counter}
+    <button onClick={Logic.Increment(-1)}>-</button>
+  </div>
+)
+```
+
+**`Logic.Increment(1)`** demonstrates how you can bind a command with some exact argument.
+
+```js
+// SearchResults.js
+...
+import * as ResultsItem from './ResultsItem';
+
+export const Logic = {
+  config({ nest }) {
+    return {
+      children: {
+        results: nest(null, ResultsItem.Logic)
+      }
+    }
+  }
+  ...
+  @Cmd.update
+  SetResultsList(ctx, results) {
+    return {
+      results: results.map(x => ResultsItem.createModel(x)),
+      loading: false
+    };
+  },
+  ...
+}
+export const View = ({ model }) => (
+  <div>
+    {model.loading && <div>Loading...</div>}
+    {model.error && <div>{model.error}</div>}
+    {model.results.map((x, i) => <ResultsItem.View model={x} />)}
+  </div>
+)
+```
+**`results: nest(null, ResultsItem.Logic)`** associates a logic with `results` model's field. But `results` is an array. In this case logic will be associated with each object of an array.
+
+**`results: results.map(x => ResultsItem.createModel(x))`** in `SetResultsList` command just creates a `ResultsItem` model for each result item.
+
+**`<ResultsItem.View model={x} />`** in shows a View of our `ResultsItem` block for each result item.
+
+## Going deeper
+What if you will have to add a user to your app? Obviously the user model should be easily accessable from anywhere of the app. For example to check is the user authorized or not, or to check a role of the user. For these cases, when you need to have a widely used model, MJS provides so called `shared block`.
```

Comment:
"What if you would need to add a user to your app?"
"...from anywhere **in** the app"
"For example, to check ***if* the user **is** authorizer or not..."
"MJS provides **the** so called `shared block`"